### PR TITLE
ENT-11400: Added junit-text-summary.sh for general use in github workflows

### PR DESCRIPTION
**Pull request by craigcomstock**
Ticket: ENT-11400
Changelog: none

### CODE REVIEW

File: ci/junit-text-summary.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+set +e
+xmlstarlet sel -T \
+  -t -i "//testcase/failure" \
+  -t -m "//testcase/failure/.." \
+  -v "//testcase/failure" -n \
+  -t -i "//testcase/failure" \
+  "$1" > "$1-failure-details.log" 2>>"$error_log"
+
+rc=$?
+if [ "$rc" = "4" ]; then
```
Comment (vpodzime) [inline ci/junit-text-summary.sh:51]: `-eq 4`

**Review by vpodzime — APPROVED** [review]
Looks good to me otherwise. Nice reusable bit! :clap

File: ci/junit-text-summary.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+if test -z "$1"; then
+  echo "Usage: $0 <junit xml file>"
+  exit 1
+fi
+if ! test -f "$1"; then
+  echo "File $1 not found"
+  exit 1
+fi
+echo "## Test Summary for $1"
+xmlstarlet sel -T \
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply ci/junit-text-summary.sh:27]: yeah, if I wrote this locally I would write `software xmlstarlet` which uses lots of different package managers (whichever is available).

The "where it is from" is in the github workflow. I mean... in most cases you get it by installing it the normal way. ;)

**Review by larsewi — APPROVED** [review]

**Review by vpodzime — APPROVED** [review]
ACK

**Comment by craigcomstock** [comment]
cherry picked, will test along-side mission-portal cherry picks :+1:

https://github.com/cfengine/buildscripts/pull/1405
https://github.com/cfengine/buildscripts/pull/1404

**Pull request merged**